### DISCUSSION TITLE
Is there a way to get a TypedDict from a basemodel?

### DISCUSSION THREAD

**Discussion by peniqliotuv** [pydantic/pydantic#2574] [Question] [answered]
In version 1.8.0, there's `create_model_from_typeddict` as documented [here](https://pydantic-docs.helpmanual.io/usage/models/#model-creation-from-namedtuple-or-typeddict) but wondering if there's a way to construct a `TypedDict` from a `BaseModel`.

**Comment by PrettyWood** [comment, marked as answer]
Hi @peniqliotuv
You could do it the other way and generate a `TypedDict` from a `BaseModel` by simply do
```py
User = TypedDict('User', **{k: v.outer_type_ for k, v in UserModel.__fields__.items()})
```
but understand that a `TypedDict` is actually interpreted as a plain `dict`. The annotations are only here for type checkers and IDE. And unfortunately tools like `mypy` don't support dynamic annotations. If you run it you'll get the error
```console
TypedDict() expects a dictionary literal as the second argument
```
So in the end I'm afraid this is pointless :/

**Comment by sobolevn** [comment]
I would  go with a custom `mypy` plugin, define a function called `model_to_typed_dict` and analyze its calls to construct a `TypedDict` based on the `BaseModel` input. It should not be too hard. 

And this is a pretty cool open-source project to write 🙂

**Reply by PrettyWood** [reply]
@sobolevn Keep us updated if you write this ;)
@peniqliotuv Can you mark this discussion as answered please?

**Reply by jd-solanki** [reply]
@sobolevn Have you written that?

**Reply by sobolevn** [reply]
Nope :)

**Reply by choucavalier** [reply]
too bad, we should do it!

**Comment by suspiciousRaccoon** [comment]


**Reply by codethief** [reply]
Well, yes, because you circumvented type checks using `cast()`.

**Comment by codethief** [comment]
It looks like `create_model_from_typeddict()` [has been removed in Pydantic v2](https://docs.pydantic.dev/latest/migration/#removed-in-pydantic-v2). Am I correct in assuming the alternative in v2 is to use a [`TypeAdapter`](https://docs.pydantic.dev/latest/concepts/type_adapter/)?

In light of this, one option _might_ be able to work entirely without Pydantic `BaseModel`s and instead work with `TypedDict`s throughout(?)
